lib/super_digit.rb

```diff
@@ -1,15 +1,23 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n)
 def super_digit(n)
-   
+  return n if n < 10
+
+  sum = 0
+  until n == 0
+    sum += n % 10
+    n = n / 10
+  end
+
+  return super_digit(sum)
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n^2)
+# Space Complexity - O(n)
 def refined_super_digit(n, k)
```

Comment:
This works but is O((dk)^2) in time complexity where d is the number of digits in n.

You can do better with dynamic programming